### PR TITLE
Org collab signup

### LINKED ISSUE #2807: Add orgs to sign-up endpoint

### PR DESCRIPTION
**Pull request by BruceMacD**
## Summary
Add orgs to Infra sign-up. This is an incremental change to unblock other orgs work (note that it is going into the `org_collab` branch, not main).

### In This Change
- 36fbd80c92394a58bf47e9ad631d925be2d1f887 Deprecate the GET `/signup` endpoint. It will always return false to allow old CLI client to continue to function (which rely on this endpoint's response).
- 103c9a3159bc5848eaf8c99cdbe5fbb3abadd3f0 Create an org on sign-up, assign an admin user, and create appropriate grants.

### Next steps 
#2801 
- Update the sign-up UI
- Update the quickstart to direct new self-hosted deployments to the "sign-up" UI
- Update ingress when a new org subdomain is created
- https://github.com/infrahq/infra/issues/2806

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [ ] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #2807

### CODE REVIEW

File: internal/access/signup.go [hunk 1]
```diff
@@ -11,50 +11,29 @@ import (
...
-
-	err := checkPasswordRequirements(db, password)
+	err = checkPasswordRequirements(db, password)
 	if err != nil {
 		return nil, err
 	}
 
-	identity := &models.Identity{Name: name}
+	identity := &models.Identity{
+		Model: models.Model{
```
Comment (BruceMacD) [inline internal/access/signup.go:32]: Is there a simpler way to set the org ID in the current implementation than this?

File: api/signup.go [hunk 2]
```diff
@@ -2,19 +2,17 @@ package api
...
 	Name     string `json:"name"`
 	Password string `json:"password"`
+	Org      string `json:"org"`
 }
 
 func (r SignupRequest) ValidationRules() []validate.ValidationRule {
 	return []validate.ValidationRule{
 		validate.Required("name", r.Name),
 		validate.Required("password", r.Password),
+		validate.Required("org", r.Org),
```
Comment (BruceMacD) [inline api/signup.go:15]: I'll separate this out into orgName and orgDomain, it's just not in this branch yet from #2779.

Comment (pdevine) [reply api/signup.go:15]: Yeah, originally I was thinking just Org Name, but this makes more sense.

File: internal/access/signup.go [hunk 3]
```diff
@@ -11,50 +11,29 @@ import (
...
-		return false, err
-	}
-
-	grants, err := data.Count[models.Grant](db.Unscoped(), data.NotPrivilege(models.InfraConnectorRole))
-	if err != nil {
-		return false, err
-	}
-
-	accessKeys, err := data.Count[models.AccessKey](db.Unscoped())
+	org := &models.Organization{Name: orgName}
```
Comment (pdevine) [inline internal/access/signup.go:20]: We should add the domain to the org model as well.

**Review by pdevine — APPROVED** [review]

**Pull request merged**